Energy threshold update for new MTD geometry

gsorrentino18 opened the pull request:
#### PR description:

The energy threshold parameter used in the MTDRecHitAlgo plugin has been modified in order to discriminate between the two different MTD ETL geometries (with one ETL disk and two ETL disks). The two MTD ETL geometries are identified via the MTDTopologyMode.

#### PR validation:

The new code has been tested in the release CMSSW_11_2_0_pre6.

gsorrentino18 committed:
Modify threshold for the new MTD ETL topology

gsorrentino18 committed:
Eliminate redundant if cycles

gsorrentino18 committed:
Minor fixes

gsorrentino18 committed:
code-check and code-format update

gsorrentino18 committed:
minor fixes

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31654/18754

- This PR adds an extra 16KB to repository

cmsbuild commented:
A new Pull Request was created by  @gsorrentino18 (Giulia Sorrentino) for master.

It involves the following packages:

RecoLocalFastTime/FTLCommonAlgos
RecoLocalFastTime/FTLRecProducers


@perrotta, @jpata, @cmsbuild, @kpedro88, @slava77 can you please review it and eventually sign? Thanks.
@fabiocos this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 18148695b4d912b18b1ed4d87cd2a15a6337d29b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/9717/summary.html
CMSSW: CMSSW_11_2_X_2020-10-02-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/9717/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2542225
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2542196
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

fabiocos commented on RecoLocalFastTime/FTLCommonAlgos/plugins/MTDRecHitAlgo.cc:
```diff
@@ -22,14 +26,20 @@ class MTDRecHitAlgo : public MTDRecHitAlgoBase {
+  double calibration_;
   const MTDTimeCalib* time_calib_;
+  const MTDTopology* topology_;
+  static constexpr int topologycode1Disk_ = 4;
```
@gsorrentino18 @casarsa there is no need to define hardcoded thresholds, you may simply use the MTDTopologyMode enum class with something like

```
if (mtdTopologyMode <= static_cast<int>(MTDTopologyMode::Mode::barphiflat)) {
```
as we are doing elsewhere in the code (and I am using in the RecoMTD/DetLayers code I am preparing)

slava77 commented:
> Reco comparison results: 4 differences found in the comparisons

<img width="783" alt="Screen Shot 2020-10-05 at 7 38 37 AM" src="https://user-images.githubusercontent.com/4676718/95093645-d8d04900-06dd-11eb-9114-b10d336d5649.png">

@smuzaffar @mrodozov 
this is not specific to this PR, but it looks like there was an update in the format if the file names, which makes inspection of the differences very hard.
the same is in https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_2_X_2020-10-02-2300+4287c1/39128/, but there it's more tolerable.
Was something changed on the server side?
Please take a look if this can be reverted back.
Thank you.

smuzaffar commented:
@slava77 , yes we migrated to a CC7 based server that might have caused this issue. I will check the apache server configuration and fix it

smuzaffar commented:
@slava77 , this should be fix now (you might have to reload the page). Can you please check and confirm if it looks good?

slava77 commented:
> @slava77 , this should be fix now (you might have to reload the page). Can you please check and confirm if it looks good?

it looks good.
Thank you for the update.

gsorrentino18 committed:
Fixing hardcoded topologymode

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31654/18797

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #31654 was updated. @perrotta, @jpata, @cmsbuild, @kpedro88, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented on RecoLocalFastTime/FTLCommonAlgos/plugins/MTDRecHitAlgo.cc:
```diff
@@ -39,6 +48,11 @@ FTLRecHit MTDRecHitAlgo::makeRecHit(const FTLUncalibratedRecHit& uRecHit, uint32
   float time = 0.;
 
+  // MTD topology
+  unsigned int index_topology = 0;  //1Disks geometry
```
it looks like this index can be computed in `getEventSetup`, no need to re-derive it for every hit

slava77 commented on RecoLocalFastTime/FTLCommonAlgos/plugins/MTDRecHitAlgo.cc:
```diff
@@ -39,6 +48,11 @@ FTLRecHit MTDRecHitAlgo::makeRecHit(const FTLUncalibratedRecHit& uRecHit, uint32
 
+  // MTD topology
+  unsigned int index_topology = 0;  //1Disks geometry
+  if (topology_->getMTDTopologyMode() > static_cast<int>(MTDTopologyMode::Mode::barphiflat))
```
why not `== btlv1etlv4` (with whatever appropriate casts). I do not see any design constraints that would prevent defining another value in `MTDTopologyMode::Mode`, which would correspond again to index 0.

Please remind me how do we get `barphiflat` vs `btlv1etlv4` vs some other value of `getMTDTopologyMode`?
It looks like a property of the full detector (mixing of different values is not possible).
Ideally, then the threshold should come from ES in a coherent way.
Having both values in the same configuration is a bit confusing, because only one can possibly be applicable.
The alternative to using ES is an appropriate Era modifier.
Only if the options above are rejected, I'd consider injecting multiple incompatible caibration/configuration parameters into the same module.

fabiocos replied:
@slava77 up to barphiflat (i.e. 4) this enum was used to distinguish various BTL scenarios, now all gone but the last one. I use the same enum now to distinguish ETL scenarios, btlv2etlv4 is the first implementation in D53, but another one is about to be stored as well. This value comes with the MTD topology object, and should be considered as a temporary tool to identify within the code which geometry is being processed, as soon as the choice will be done this will not be needed any more.

In the specific case of this parameter, there is a change in thickness of the silicon detector in the change of scenarios that justify this move, and this change will not be a function of the new scenarios (i.e. LGAD thickness should be fixed now regardless of the geometrical arrangement of modules on the discs' faces). So for what we know now, this selection is ok. Your suggestion, if I understand correctly, would be to create an EStype object that provides these parameters as a function of the topology mode, do I understand it correctly? @casarsa @gsorrentino18 do you see other parameters that need to be tuned?

slava77 replied:
> up to barphiflat (i.e. 4) this enum was used to distinguish various BTL scenarios, now all gone but the last one. I use the same enum now to distinguish ETL scenarios, btlv2etlv4 is the first implementation in D53, but another one is about to be stored as well.

I'm not sure if this historical precedent can suggest that in the future any new enum will also be about ETL 2-disk cases; that's why I think that selection of values explicitly is more appropriate than a range comparison `>`.

> Your suggestion, if I understand correctly, would be to create an EStype object that provides these parameters as a function of the topology mode, do I understand it correctly?

That would be an ideal case; it's probably not much complicated because at least in this case the related data is already handled via ES.
An era modifier is another option.

casarsa replied:
> @casarsa @gsorrentino18 do you see other parameters that need to be tuned?

@fabiocos In principle, yes, the calibration constant (the ETL energy is converted into MIP units in the simulation and the MIP peak value depends on the thickness of the sensor), but we are planning on tackling that in another dedicated PR that modifies also the simulation code.
As for this PR, the calibration constant is an internal parameter (and currently kind of arbitrary) and as long as its value is consistent between simulation and reconstruction the result is independent of its actual value.

gsorrentino18 replied:
> That would be an ideal case; it's probably not much complicated because at least in this case the related data is already handled via ES.
> An era modifier is another option.

Following your suggestions, we are indeed thinking about implementing an Era Modifier to provide the parameters as a function of the geometry.

gsorrentino18 replied:
@slava77 Before modifying the PR, if I understand well, do you want me to implement a modifier to the Phase2 Era similar to those listed here? https://cmssdt.cern.ch/lxr/source/Configuration/Eras/python/Era_Phase2_cff.py?v=CMSSW_11_2_X
Or were you suggesting something different?

slava77 replied:
> @slava77 Before modifying the PR, if I understand well, do you want me to implement a modifier to the Phase2 Era similar to those listed here? https://cmssdt.cern.ch/lxr/source/Configuration/Eras/python/Era_Phase2_cff.py?v=CMSSW_11_2_X
> Or were you suggesting something different?

yes, that's what I was looking for; I guess you pick the MTD-related name for the modifier and then we see (jointly if it makes sense)
It will be in Era_Phase2CXX_cff file @kpedro88 please suggest a file for this

kpedro88 replied:
@gsorrentino18 @slava77 it depends on which geometry(s) it covers.

Right now we mainly have top-level Phase2 Eras depending on the calorimeter version (`Phase2C9`, `Phase2C10`, `Phase2C11`, `Phase2C12`). Based on @fabiocos' comments below, the new top-level Era should probably be `Phase2C11I13` (since I13 is the new MTD geometry in #31710). To be concrete, it should start from `Phase2C11` and then add a new modifier like `phase2_etl_2disc`. But MTD experts should correct me if I'm wrong about the context.

fabiocos replied:
@kpedro88 @slava77 I am about to submit a PR with the last piece needed to run a whole production workflow with the  new ETL (2 discs), which needs the updates. in this PR (as the thickness of the silicon is now realistically the 50 microns of the LGAD). When this enters in production, and geometry for HGCal provided by @bsunanda should go with it, so this modifier should become in my view an integral part of the baseline Era. I see the special version as a short term interim solution before moving to that new regime, as I expect that the review of the new code will take a bit of time

kpedro88 replied:
@fabiocos @gsorrentino18 is the suggestion to go ahead with this PR without introducing a new modifier?
if so, there are still a few other unanswered comments; please address these

gsorrentino18 replied:
@kpedro88 My intention is to implement an Era Modifier, but since this PR is related to #31765 and #31769, which introduce bigger modifications in the code, I though it would be better to hold this PR until the review of the other two is completed.

fabiocos replied:
@kpedro88 @slava77 @bsunanda so what is the preferred strategy? Go for a short period with an intermediate Era and then move the ETL definition into a new Phase2CXX default? Or we may keep this PR on hold and use it for tests of #31765 with #31769, add an Era there and then merge everything in one shot. @silviodonato @qliphy please clarify the preferred strategy

silviodonato replied:
@fabiocos I'm ok with having an intermediate Era

cmsbuild commented:
+1
Tested at: cb5659820435f251e7a17cd5fbacce3340aa4729
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/9746/summary.html
CMSSW: CMSSW_11_2_X_2020-10-05-1200
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/9746/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2542225
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2542202
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

kpedro88 commented on RecoLocalFastTime/FTLCommonAlgos/plugins/MTDRecHitAlgo.cc:
```diff
@@ -3,12 +3,16 @@
 
+#include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+#include "Geometry/MTDCommonData/interface/MTDTopologyMode.h"
+#include <iostream>
```
this include should not be needed, because cout is not allowed.

slava77 commented:
@gsorrentino18 @fabiocos @kpedro88 
what's a good workflow number to test the new variant of MTD?
IIUC, the default jenkins tests are not using it.

fabiocos commented:
@slava77 up to now you might use D53, but this is not available (yet) in the standard matrix because of the RECO geometry not being updated. I have a branch ready for this, but I am trying to improve it performance-wise before making a PR. Anyway I think I will move forward quite soon

fabiocos commented:
@slava77 after the initial test, we may use test wf.s 33034 and 33434 in #31710 . Any update of eras should be propagated there

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#31769,cms-sw/cmssw#31765

cmsbuild commented:
-1

Tested at: cb5659820435f251e7a17cd5fbacce3340aa4729

CMSSW: CMSSW_11_2_X_2020-10-13-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-886545/9915/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation warning while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```
 See details on the summary page.

cmsbuild commented:
Comparison not run due to Build errors/Fireworks only changes/No short matrix requested (RelVals and Igprof tests were also skipped)

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-886545/9918/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-886545/33034.0_TTbar_14TeV+2026D72+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-886545/33434.0_TTbar_14TeV+2026D73+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2543353
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2543324
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1459.401 KiB( 34 files compared)
 - DQMHistoSizes: changed ( 23234.0,... ): 481.180 KiB      MTD/ETL
 - DQMHistoSizes: changed ( 23234.0,... ): 5.287 KiB        MTD/GlobalReco
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: cb5659820435f251e7a17cd5fbacce3340aa4729
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/9938/summary.html
CMSSW: CMSSW_11_2_X_2020-10-13-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/9938/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2543752
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2543727
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

fabiocos commented:
@gsorrentino18 @jfernan2 no ETL change in this PR (@silviodonato but it looks like there is some instability in the tests)

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#31765
Test Parameters:
  - **MATRIX_EXTRAS** = 33434.0,33034.0

silviodonato commented:
> @gsorrentino18 @jfernan2 no ETL change in this PR (@silviodonato but it looks like there is some instability in the tests)

@fabiocos are you referring to the differences in `SiOuterTrackerV/Tracks/FinalResolution/d0Resolution` ?
![image](https://user-images.githubusercontent.com/6177433/95984875-25293080-0e24-11eb-902b-bb3f9c327ba5.png)

fabiocos commented:
@silviodonato that looks the only significant difference

cmsbuild commented:
+1
Tested at: cb5659820435f251e7a17cd5fbacce3340aa4729
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f32b79/9943/summary.html
CMSSW: CMSSW_11_2_X_2020-10-13-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

perrotta commented:
@fabiocos @silviodonato I noticed those supposed differences in the PR test outputs since a few days already, and they are not reproducible: then can disappear if the tests are repeated (see for example #31724, which is the first time I saw them)

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f32b79/9943/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-f32b79/33034.0_TTbar_14TeV+2026D72+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /build/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-f32b79/33434.0_TTbar_14TeV+2026D73+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2543752
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2543727
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

jfernan2 commented:
> > @gsorrentino18 @jfernan2 no ETL change in this PR (@silviodonato but it looks like there is some instability in the tests)
> 
> @fabiocos are you referring to the differences in `SiOuterTrackerV/Tracks/FinalResolution/d0Resolution` ?
> ![image](https://user-images.githubusercontent.com/6177433/95984875-25293080-0e24-11eb-902b-bb3f9c327ba5.png)

Indeed the change is only in error bar sizes

jfernan2 commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#31765
Test Parameters:
  - **MATRIX_EXTRAS** = 33434.0,33034.0

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmssw#31765
Test Parameters:
  - **MATRIX_EXTRAS** = 33434.0,33034.0

cmsbuild commented:
+1
Tested at: cb5659820435f251e7a17cd5fbacce3340aa4729
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f32b79/10120/summary.html
CMSSW: CMSSW_11_2_X_2020-10-19-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-f32b79/10120/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-f32b79/33034.0_TTbar_14TeV+2026D72+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-f32b79/33434.0_TTbar_14TeV+2026D73+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2544100
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2544077
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

gsorrentino18 committed:
commit d599c6e7a72ae9399d67f12cda4c03f52f7bf6f3
Author: Giulia Sorrentino <sorrentinogiulia10@gmail.com>
Date:   Tue Oct 20 18:40:16 2020 +0200

    adding Era_Phase2C11_etlv4

commit f25000eac410308fcf5c2a6611fe7fb9313d286c
Author: Giulia Sorrentino <sorrentinogiulia10@gmail.com>
Date:   Wed Oct 21 13:22:24 2020 +0200

    Restoring MTDRecHitAlgo and moving etlV4 modifier from ProcessModifiers to Eras

Adding a new Era to provide energy threshold and calibration parameters as a function of MTD geometry

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31654/19280

- This PR adds an extra 28KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/StandardSequences/python/Eras.py  modified in PR(s): #31726

cmsbuild commented:
Pull request #31654 was updated. @perrotta, @civanch, @silviodonato, @mdhildreth, @cmsbuild, @franzoni, @kpedro88, @slava77, @jpata, @qliphy, @fabiocos, @davidlange6 can you please check and sign again.

fabiocos commented:
@slava77 @bsunanda the latest commit by @gsorrentino18 adds the updated Phase2C11 era that we had discussed, this should go in the direction of the update suggested. At this point I should update #31765 to use this new Era for both D72 and D73

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 commented on Configuration/Eras/python/Era_Phase2C11_etlV4_cff.py:
```diff
@@ -0,0 +1,6 @@
+from Configuration.Eras.Era_Phase2C11_cff import Phase2C11
+from Configuration.Eras.Modifier_phase2_etlV4_cff import phase2_etlV4
+
+Phase2C11_etlV4 = cms.ModifierChain(Phase2C11, phase2_etlV4)
```
for top-level Eras, please follow the existing naming scheme:
e.g. C11 is a subdetector version for HGCal described in https://github.com/cms-sw/cmssw/tree/master/Configuration/Geometry, so use the corresponding subdetector version for MTD

fabiocos replied:
@kpedro88 as discussed above, there is no intention to add an MTD dedicated era, this is a temporary workaround while #31765 is integrated to have a variant of Phase2C11  where D72 and D73 can be exercised without problems. When that PR is integrated, there is no reason to have scenarios with PhaseC11 AND old ETL , Phase2C11 should include this modifier as a new default, and the era added here should just be retired. Do you see a problem with this? According to what I discussed with @bsunanda Phase2C11 should stay for a while, while the old ETL in conjunction with it should disappear asap

kpedro88 replied:
@fabiocos thanks for clarifying. in this case, the temporary Era is acceptable as it currently is.

fabiocos replied:
@kpedro88 thanks, so I will move forward accordingly in #31765, also to make a meaningful performance test as required by @jpata . Without #31654 , there are very few hits in ETL and any performance test would be biassed

cmsbuild commented:
-1

Tested at: 98c389fcc3f9def16485cc60e17369c39b435ce6

CMSSW: CMSSW_11_2_X_2020-10-20-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/10176/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test TestDQMOnlineClient-beam_dqm_sourceclient had ERRORS

cmsbuild commented:
Comparison job queued.

fabiocos commented:
@qliphy @silviodonato the unit test error seems to me unrelated to this PR...

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4287c1/10176/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2544100
 - DQMHistoTests: Total failures: 7
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2544071
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

slava77 commented on RecoLocalFastTime/FTLRecProducers/python/mtdRecHits_cfi.py:
```diff
@@ -13,6 +13,9 @@
 )
 
+from Configuration.Eras.Modifier_phase2_etlV4_cff import phase2_etlV4
+phase2_etlV4.toModify(_endcapAlgo, thresholdToKeep = 0.005 )
+phase2_etlV4.toModify(_endcapAlgo, calibrationConstant = 0.001 )
```
```suggestion
phase2_etlV4.toModify(_endcapAlgo, thresholdToKeep = 0.005, calibrationConstant = 0.001 )
```
is less verbose

slava77 commented on RecoLocalFastTime/FTLRecProducers/python/mtdRecHits_cfi.py:
```diff
@@ -23,3 +26,5 @@
     BarrelHitsName = cms.string('FTLBarrel'),
     EndcapHitsName = cms.string('FTLEndcap'),
 )
+
+
```
```suggestion
```
not necessary

slava77 reviewed (commented):
IIUC, there is a preference now to have this PR be merged in #31765 
based on https://github.com/cms-sw/cmssw/pull/31765#issuecomment-713700013

the comments are minor enough to be ignored; still, if there is a chance to update, please pick these up

fabiocos commented:
@slava77 thanks for the comments, I will integrate them in #31765

fabiocos commented:
@slava77 your comments have been addressed in https://github.com/cms-sw/cmssw/pull/31765/commits/ee6ac209e3c5e14700a03f093588c4952049a509 please have a look in case

fabiocos commented:
@gsorrentino18 unless @slava77 @kpedro88 @silviodonato have a different advice, I would say that at this point this PR may just be closed

slava77 commented:
-1

superseded by #31765

kpedro88 commented:
please close
